gridstatus/tests/source_specific/test_eia.py

```diff
@@ -426,6 +431,65 @@ def test_get_henry_hub_natural_gas_spot_prices_historical_date_range():
     assert df["Interval End"].max() == pd.Timestamp("2024-01-03", tz=HENRY_HUB_TIMEZONE)
 
 
+def _check_generators_data(
```

Comment:
Refactored the common checks into one method instead of repeating them in each test.